### ISSUE TITLE
Creating aws.elasticache.ServerlessCache resource encoder exception

### ISSUE THREAD

**Issue by davidfries** [pulumi/pulumi-aws#3428]
### What happened?

When creating a new aws.elasticache.ServerlessCache resource, there is an exception thrown:

```
 error: Failed to prepare a resource encoder: cannot derive an encoder for resource "aws_elasticache_serverless_cache": Cannot derive an encoder for property "endpoint": issue deriving an object encoder: expected an object type schema at walk.NewSchemaPath().GetAttr("endpoint").Element()
```

### Example

```
export class Redis extends pulumi.ComponentResource {
  public readonly redisCluster: aws.elasticache.ServerlessCache;
  constructor(
    name: string,
    params: RedisParams,
    tags: BaseTags,
    opts?: pulumi.ComponentResourceOptions,
  ) {
    super(`[redacted]:Redis:${params.clusterName}`, name, {}, opts);
    this.redisCluster = new aws.elasticache.ServerlessCache(
      `${name}-cluster`,
      {
        engine: "redis",
        tags: tags,
        name: params.clusterName,
        dailySnapshotTime: "05:00",
      },
      { parent: this },
    );
  }
}
```

### Output of `pulumi about`

```
CLI          
Version      3.104.2
Go Version   go1.21.6
Go Compiler  gc

Plugins
NAME        VERSION
aws         6.20.1
awsx        2.5.0
cloudflare  5.20.0
docker      4.5.1
docker      3.6.1
nodejs      unknown

Host     
OS       debian
Version  bookworm/sid
Arch     x86_64

This project is written in nodejs: executable='/home/linuxbrew/.linuxbrew/bin/node' version='v21.6.1'

Dependencies:
NAME                VERSION
zod                 3.22.4
@pulumi/aws         6.20.1
@pulumi/awsx        2.5.0
@pulumi/cloudflare  5.20.0
@pulumi/pulumi      3.104.2
@types/node         18.19.14
prettier            3.2.5
typescript          4.9.5
```

### Additional context

This resource is being created in a component resource, which is being instantiated inside another component resource.

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by iwahbe** [comment]
Hi @davidfries. Thanks for letting us know about this.

**Comment by davidfries** [comment]
@iwahbe sorry for the delay. I can confirm it's fixed in `@pulumi/aws 6.22.2`
